tests/validation/validation_data.cpp

```diff
@@ -15,6 +16,38 @@ namespace mc {
 
 trace_io g_trace_io;
 
+util::path trace_io::find_datadir() {
+    // If environment variable is set, use that in preference.
+    if (const char* env_path = std::getenv("NMC_DATADIR")) {
+        return util::path(env_path);
```

Comment:
There's not much we can do if it's not a directory; second-guessing the user here and substituting something else would lead to obscure issues. Better that it just not find the data IMO.